File: app/src/main/java/com/worldventures/dreamtrips/wallet/service/nxt/model/MultiRequestBody.java
package com.worldventures.dreamtrips.wallet.service.nxt.model;

import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import java.util.List;

@Gson.TypeAdapters
@Value.Immutable
public interface MultiRequestBody {

   @Nullable
   @SerializedName("SessionToken")
   String sessionToken();

   @SerializedName("MultiRequestElement")
   List<MultiRequestElement> multiRequestElements();
}

File: wallet/src/main/java/com/worldventures/wallet/service/nxt/model/MultiRequestBody.kt
package com.worldventures.wallet.service.nxt.model

import com.google.gson.annotations.SerializedName

data class MultiRequestBody(
      @field:SerializedName("MultiRequestElement") val multiRequestElements: List<MultiRequestElement>,
      @field:SerializedName("SessionToken") val sessionToken: String? = null

)
